File: Session7/C++/FirstUniqueCharInString.cpp
https://leetcode.com/problems/first-unique-character-in-a-string/

class Solution {
public:
    int firstUniqChar(string s) {
        
        unordered_map<char, int> charFrequency;
        
        for (char c : s) {
            charFrequency[c]++;
        }
        
        for (int i=0; i<s.size(); i++) {
            if (charFrequency[s[i]] == 1) {
                return i;
            }
        }
        
        return -1;
    }
};

File: Session7/Java/FirstUniqueCharInString.java
https://leetcode.com/problems/first-unique-character-in-a-string/

class Solution {
    public int firstUniqChar(String s) {
        int n = s.length();
        Map<Character, Integer> charFrequency = new HashMap<>();
        
        for (int i=0; i<n; i++) {
            char c = s.charAt(i);
            charFrequency.put(c, charFrequency.getOrDefault(c, 0) + 1);
        }
        
        for (int i=0; i<n; i++) {
            if (charFrequency.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        
        return -1;
    }
}